### ISSUE TITLE
[Bug] Grass 7.8.4 does not start on Debian Testing: OnInit returned false, exiting...

### ISSUE THREAD

**Issue by rduivenvoorde** [OSGeo/grass#1134]
Trying to start Grass on my Debian Testing desktop fails with message below.

I removed the an earlier 'grass' dir (locations dir?), and a .grass7 dir I had floating around in my homedir.

I also created a fresh user to be sure no old stuff was interferring in my own environment.

All lead to the same error.
using `g.gui wxpython` in the grass console results in the same err

Any hints to start grass appreciated. :-)
Thanks

```
[richard@oost ~]$ grass
Starting GRASS GIS...
Traceback (most recent call last):
  File "/usr/lib/grass78/gui/wxpython/gis_set.py", line 562, in OnWizard
    defineRegion = RegionDef(self, location=gWizard.location)
  File "/usr/lib/grass78/gui/wxpython/location_wizard/dialogs.py", line 40, in __init__
    wx.Dialog.__init__(self, parent, id, title, size=size)
TypeError: Object(): arguments did not match any overloaded call:
  overload 1: too many arguments
  overload 2: too many arguments
Cleaning up temporary files...

          __________  ___   __________    _______________
         / ____/ __ \/   | / ___/ ___/   / ____/  _/ ___/
        / / __/ /_/ / /| | \__ \\_  \   / / __ / / \__ \
       / /_/ / _, _/ ___ |___/ /__/ /  / /_/ // / ___/ /
       \____/_/ |_/_/  |_/____/____/   \____/___//____/

Welcome to GRASS GIS 7.8.4
GRASS GIS homepage:                      https://grass.osgeo.org
This version running through:            Bash Shell (/bin/bash)
Help is available with the command:      g.manual -i
See the licence terms with:              g.version -c
See citation options with:               g.version -x
If required, restart the GUI with:       g.gui wxpython
When ready to quit enter:                exit

Launching <wxpython> GUI in the background, please wait...
GRASS 7.8.4 (world):~ > Traceback (most recent call last):
  File "/usr/lib/grass78/gui/wxpython/wxgui.py", line 105, in OnInit
    mainframe = GMFrame(parent=None, id=wx.ID_ANY,
  File "/usr/lib/grass78/gui/wxpython/lmgr/frame.py", line 129, in __init__
    self._menuTreeBuilder = LayerManagerMenuData(message_handler=add_menu_error)
  File "/usr/lib/grass78/gui/wxpython/lmgr/menudata.py", line 40, in __init__
    MenuTreeModelBuilder.__init__(
  File "/usr/lib/grass78/gui/wxpython/core/menutree.py", line 71, in __init__
    xmlTree = etree.parse(filename)
  File "/usr/lib/python3.8/xml/etree/ElementTree.py", line 1202, in parse
    tree.parse(source, parser)
  File "/usr/lib/python3.8/xml/etree/ElementTree.py", line 595, in parse
    self._root = parser._parse_whole(source)
xml.etree.ElementTree.ParseError: no element found: line 1, column 0
OnInit returned false, exiting...
^C
GRASS 7.8.4 (world):~ > 
exit
WARNING: Failed to start shell '/bin/bash'
Cleaning up temporary files...
Done.

Goodbye from GRASS GIS

```

**Comment by neteler** [comment]
I believe that this is a duplicate of #1119 which has already been fixed in #1120 - scheduled for GRASS GIS 7.8.5.

If you don't want to wait for that, you may consider to apply these few changes locally:
https://github.com/OSGeo/grass/pull/1120/files

**Comment by rduivenvoorde** [comment]
@neteler thanks, sorry for duplicating it. I tried to apply changes, but I do not have an update_menudata.py

And without that change I still have the same issue. Creating a tools dir with that file does not help either :-(

**Comment by neteler** [comment]
You may simply download the weekly source code snapshot from here (I have just triggered the cronjob on the server to force an update):

- https://grass.osgeo.org/grass78/source/snapshot/

or clone the repo and switch to https://github.com/OSGeo/grass/tree/releasebranch_7_8
- `git clone https://github.com/OSGeo/grass `

**Comment by nilason** [comment]
@rduivenvoorde What python version do you have?
try: `python3 -c "import sys, wx; print(sys.version); print(wx.version())"`

**Comment by rduivenvoorde** [comment]
@nilason 
```
$ python3 -c "import sys, wx; print(sys.version); print(wx.version())"
3.8.6 (default, Sep 25 2020, 09:36:53) 
[GCC 10.2.0]
4.0.7 gtk3 (phoenix) wxWidgets 3.0.5

```

**Comment by nilason** [comment]
Interesting, the fixes in #1120 are supposed to address changes with Python 3.9. Yet, similar problem is seemingly occurring with 3.8.6 for you.

You don't incidentally have $GRASS_PYTHON set?
`echo $GRASS_PYTHON`
`$GRASS_PYTHON -c "import sys, wx; print(sys.version); print(wx.version())" `

**Comment by rduivenvoorde** [comment]
Nope no $GRASS_PYTHON

```
$ echo $GRASS_PYTHON
(nothing)
```

**Comment by mlennert** [comment]
I can confirm the issue with the same Debian testing setup as Richard.

IIUC, the error actually comes from the fact that menudata.xml is empty. And it is empty because of the mentioned bugs. However, the tools directory is not shipped in the Debian package and so a user cannot update that file automatically (also because it is owned by root).

Some ideas to maybe make things easier for users:

- Could the code in update_menudata.py be wrapped in a module that users can run to create a menudata.xml file in their .grass7 directory ?

- At least we could introduce a check for whether menudata.xml is empty and if it is, provide an understandable message and/or creates some default menu ?

**Comment by mlennert** [comment]
If you have root access, a quick solution is:

- Download binary snapshot for Linux: https://grass.osgeo.org/grass78/binary/linux/snapshot/
- Extract from that snapshot the two files menudata.xml and module_tree_menudata.xml (in gui/wxpython/xml/)
- As root: copy the two files into /usr/lib/grass78/gui/wxpython/xml

**Comment by rduivenvoorde** [comment]
@mlennert Yes that helps! Thanks.

Is this a code issue? 
Or a Debian packaging issue? I'm happy to create a bugreport in the Debian issue tracker, if needed.
I will also sent an email/ping @sebastic who packages grass for Debian (I think).

**Comment by mlennert** [comment]
It is a code issue, that was resolved and now calls for a repackaging, but maybe there should be a quick 7.8.6 release with that fix, unless Bas is willing to cherry pick the fix and repackage 7.8.4. @neteler what do you think ?

**Comment by sebastic** [comment]
https://salsa.debian.org/debian-gis-team/grass/-/commit/8fc328b567b4a2904106c0b6bc2aa3434d190cdc

Testing migration will likely be blocked by [python3-defaults](https://qa.debian.org/excuses.php?package=python3-defaults), so testing users will need to wait or get it from unstable.

**Comment by neteler** [comment]
I am fine with a quick 7.8.5 (which should then also include the new g.extension fix)

**Comment by petrasovaa** [comment]
> It is a code issue, that was resolved and now calls for a repackaging, but maybe there should be a quick 7.8.6 release with that fix, unless Bas is willing to cherry pick the fix and repackage 7.8.4. @neteler what do you think ?

Could someone explain how does the fix in #1120 influences this? In this case we have Python 3.8, this fix was needed because some functions were removed from 3.9, so I don't understand what is happening here, as @nilason also notes above.

**Comment by rduivenvoorde** [comment]
@petrasovaa (I'm just guessing here) maybe the build of the package was done with 3.9? Hence I had no menudata.xml (and cannot recreate it). Running with 3.8 fails because of the empty menudata.xml, not so much because of the code fixes not working... I think...

**Comment by landam** [comment]
> I believe that this is a duplicate of #1119 which has already been fixed in #1120 - scheduled for GRASS GIS 7.8.5.

I can confirm 7.8.4 bug (also using Debian testing). Locally compiled 7.8.5dev works fine. Can we close this issue?

**Comment by rduivenvoorde** [comment]
Yep, sure fine. Hope it will land in Debian soon.
